tests/unit_tests/test_unit.py

```
@@ -1661,6 +1661,59 @@ def test_get_candidates_isotope_error(tiny_fasta_file):
+        for record in caplog.records
+    )
+
+
+def test_psm_batches_empty_candidates(tiny_config):
+    """Test that _psm_batches returns nothing when no candidates exist."""
+    tokenizer = depthcharge.tokenizers.peptides.PeptideTokenizer(
+        residues=Config(tiny_config).residues
+    )
+    db_model = DbSpec2Pep(tokenizer=tokenizer)
+    db_model.protein_database = unittest.mock.MagicMock()
+    db_model.protein_database.get_candidates = lambda mz, charge: pd.Series(
```

Ruff (0.15.7)

[warning] Unused lambda argument: `mz`

(ARG005)